quarkus/tests/junit5/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java

```diff
@@ -340,7 +340,7 @@ private void threadDump() {
         try {
             ProcessBuilder builder = new ProcessBuilder("kill", "-3", String.valueOf(keycloak.pid()));
             Process p = builder.start();
-            p.onExit().get(getStartTimeout(), TimeUnit.MILLISECONDS);
+            p.onExit().get(3, TimeUnit.MINUTES);
```

Comment:
This is IMHO missing a comment why you whose 3 minutes here. Otherwise people will wonder in the future why this is here. 

Looking at `application.properties`, the transaction timeout is 300 seconds, which is 5 minutes. 

Waiting for 3 minutes here, and (given your comment in the issue) waiting for 2 minutes at another place, would add up to 5 minutes, which matches the 300 seconds. Maybe it would be good to add one more minute to he on the safe side. Whatever your reasoning is here, state the "why" for those minutes you choose here in a comment.

Reply:
Here I wrongly increased the timeout on the kill command while we actually need to increase the stop command. The reason is the mentioned Narayana bug, which makes the process hang longer. I already made the change, I also added a comment that the timeout can be reverted once the bug is addressed.

Speaking about the timeout length, 180 seconds should be just enough. From the perspective of other tests, the used .waitFor() will continue as soon as the process terminates. This way the other tests shouldn't be influenced.

Reply:
@Pepo48 - I see that the code has changed, but I'd still like to see a comment in the code detailing the "why" you chose 180 seconds so our future selfs and other will know. 

I someone wants to comment why such a comment is not necessary, I'm open to hear that.

Reply:
This is in any case just a workaround that should go away once https://github.com/jbosstm/narayana/pull/2201 is resolved.

Couldn't find an issue to remove the workaround. @shawkins Did we create one, I'm not sure. :)

Reply:
> Couldn't find an issue to remove the workaround. @shawkins Did we create one, I'm not sure. :)

Not yet, just the comments in the code about the upstream issue.

Reply:
Created #26454

Reply:
@ahus1 As far as I understand, the 180 seconds is just an "educated guess" (as with anything when dealing with async stuff in testsuite). Do you want to add a comment with that? :)

Reply:
I was hoping something different than an educated guess, let's leave it as is.